Test `ZeebePartitionTest.shouldReportUnhealthyIfTransitionStepIsStuck` is unstable

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? 
   - Once
- Does it block your work?
  - No, but it decreases confidence in the CI. 
- Do we suspect that it is a real failure?
  - Not sure. 

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
Error:  Tests run: 23, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.838 s <<< FAILURE! -- in io.camunda.zeebe.broker.system.partitions.ZeebePartitionTest
Error:  io.camunda.zeebe.broker.system.partitions.ZeebePartitionTest.shouldReportUnhealthyIfTransitionStepIsStuck -- Time elapsed: 0.073 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual:
  "Transition from null on term 0 appears blocked, step null has been running for PT2M0.013S"
to contain:
  "Transition from FOLLOWER on term 0 appears blocked" 
	at io.camunda.zeebe.broker.system.partitions.ZeebePartitionTest.shouldReportUnhealthyIfTransitionStepIsStuck(ZeebePartitionTest.java:419)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

-  [Downloaded CI logs](https://drive.google.com/file/d/1W_ZsO04nsEU7qKXhsLkU9g-S2YzdzGt_/view?usp=sharing)

--- a/zeebe/scheduler/src/test/java/io/camunda/zeebe/scheduler/testing/ControlledActorSchedulerRule.java
+++ b/zeebe/scheduler/src/test/java/io/camunda/zeebe/scheduler/testing/ControlledActorSchedulerRule.java
@@ -7,6 +7,8 @@
  */
 package io.camunda.zeebe.scheduler.testing;
 
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
 import io.camunda.zeebe.scheduler.Actor;
 import io.camunda.zeebe.scheduler.ActorScheduler;
 import io.camunda.zeebe.scheduler.ActorScheduler.ActorSchedulerBuilder;
@@ -20,6 +22,8 @@ import io.camunda.zeebe.scheduler.clock.ControlledActorClock;
 import io.camunda.zeebe.scheduler.future.ActorFuture;
 import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
 import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
 import org.agrona.concurrent.IdleStrategy;
 import org.junit.rules.ExternalResource;
 
@@ -50,7 +54,11 @@ public final class ControlledActorSchedulerRule extends ExternalResource {
 
   @Override
   protected void after() {
-    actorScheduler.stop();
+    try {
+      actorScheduler.stop().get(5000, MILLISECONDS);
+    } catch (final InterruptedException | ExecutionException | TimeoutException e) {
+      throw new RuntimeException(e);
+    }
   }
 
   public ActorFuture<Void> submitActor(final Actor actor) {

--- a/zeebe/scheduler/src/test/java/io/camunda/zeebe/scheduler/testing/ControlledActorThread.java
+++ b/zeebe/scheduler/src/test/java/io/camunda/zeebe/scheduler/testing/ControlledActorThread.java
@@ -12,13 +12,12 @@ import io.camunda.zeebe.scheduler.ActorThreadGroup;
 import io.camunda.zeebe.scheduler.ActorTimerQueue;
 import io.camunda.zeebe.scheduler.TaskScheduler;
 import io.camunda.zeebe.scheduler.clock.ActorClock;
-import java.util.concurrent.BrokenBarrierException;
-import java.util.concurrent.CyclicBarrier;
-import org.agrona.LangUtil;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Phaser;
 import org.agrona.concurrent.IdleStrategy;
 
 public final class ControlledActorThread extends ActorThread {
-  private final CyclicBarrier barrier = new CyclicBarrier(2);
+  private final Phaser phaser = new Phaser(2);
 
   public ControlledActorThread(
       final String name,
@@ -32,17 +31,18 @@ public final class ControlledActorThread extends ActorThread {
     this.idleStrategy = new ControlledIdleStrategy(idleStrategy);
   }
 
-  public void resumeTasks() {
-    try {
-      barrier.await(); // work at least 1 full cycle until the runner becomes idle after having been
-    } catch (final InterruptedException | BrokenBarrierException e) {
-      LangUtil.rethrowUnchecked(e);
-    }
+  @Override
+  public CompletableFuture<Void> close() {
+    phaser.arriveAndDeregister();
+    return super.close();
   }
 
-  // ControlledActorThread#resumeTasks must be called before this method
+  public void resumeTasks() {
+    phaser.arriveAndAwaitAdvance();
+  }
+
   public void waitUntilDone() {
-    while (barrier.getNumberWaiting() < 1) {
+    while (phaser.getArrivedParties() < 1) {
       // spin until thread is idle again
       Thread.yield();
     }
@@ -62,12 +62,7 @@ public final class ControlledActorThread extends ActorThread {
     @Override
     protected void onIdle() {
       super.onIdle();
-
-      try {
-        barrier.await();
-      } catch (final InterruptedException | BrokenBarrierException e) {
-        LangUtil.rethrowUnchecked(e);
-      }
+      phaser.arriveAndAwaitAdvance();
     }
   }
 }

